web/profiles/custom/os2loop/modules/os2loop_flag_content/src/Form/FlagContentForm.php

```diff
@@ -167,16 +162,20 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $key = 'flag_content';
     $params['reason'] = $form_state->getValue('reason');
     $params['message'] = $message;
-    $params['node_title'] = $node->label();
+    $params['node'] = $node;
     $langcode = $this->currentUser->getPreferredLangcode();
     $send = TRUE;
     $result = $this->mailManager->mail($module, $key, $to, $langcode, $params, NULL, $send);
     if ($result['result'] !== TRUE) {
-      $this->messenger->addMessage($this->t('There was a problem sending your message and it was not sent.'), 'error');
+      $this->messenger->addError($this->t('There was a problem sending your message and it was not sent.'));
     }
     else {
-      $this->messenger->addError($this->t('Your message has been sent.'));
+      $this->messenger->addStatus($this->t('Your message has been sent.'));
     }
+
+    $redirectUrl = Url::fromRoute('entity.node.canonical', ['node' => $node->id()])->toString();
+    $response = new RedirectResponse($redirectUrl);
+    $response->send();
```

Comment:
This should (probably) be moved into the `else` branch, i.e. only be triggered when the message is sent successfully.